### DISCUSSION TITLE
Allow us to not raise saving events for sorting nodes through Contentservice

### DISCUSSION THREAD

**Discussion by jemayn** [umbraco/Umbraco-CMS#15630] [Features and ideas]
I have a case where the user wants new nodes in a tree to automatically be added to the top of the parents sortorder.

I thought I could just add a ContentSavedNotification that does something like this:

```csharp
 // Only trigger for new nodes - https://docs.umbraco.com/umbraco-cms/reference/notifications/determining-new-entity
var dirty = (IRememberBeingDirty)savedEntity;
var isNew = dirty.WasPropertyDirty("Id");
if(!isNew) continue;

var siblings = _contentService.GetPagedChildren(savedEntity.ParentId, 0, 100, out long totalRecords);
var sortedSiblings = siblings.Where(sibling => sibling.Id != savedEntity.Id).ToList();
sortedSiblings.Insert(0, savedEntity);

_contentService.Sort(sortedSiblings);
```

However, the contentService.Sort method: https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Core/Services/ContentService.cs#L2849

Ends up sending another saving event: 
https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Core/Services/ContentService.cs#L2951

So if we have other contentSaving notification handlers then they are triggered for all nodes in the tree.
Not sure when it changed, but in the past it was optional if we wanted to raise events or not: https://github.com/umbraco/Umbraco-CMS/blob/release-8.18.12/src/Umbraco.Core/Services/Implement/ContentService.cs#L2338

So would it be possible to add that back? Or atleast have some sort of eventmessages we could look at that would tell us it was from the sorting in the contentsaving event? - As right now they are empty:
![image](https://github.com/umbraco/Umbraco-CMS/assets/79840720/5cb89ef4-a857-4e21-8bf3-4a3ec3528efd)

**Comment by bergmania** [comment]
You can suppress notifications on your scope.

```cs
        using var scope = _coreScopeProvider.CreateCoreScope();
        using(scope.Notifications.Suppress()){
             _contentService.Sort(sortedSiblings);
        }
        scope.Complete();

        }
```

**Reply by jemayn** [reply]
Thanks Bjarke!

Couldn't find anything of that but it is great news.
Will make a PR to the docs when I find a few minutes to add a mention of this 🙂
